### PR TITLE
feat: new format option 'summary'

### PR DESCRIPTION
**Pull request by RomanHotsiy**
## What/Why/How?

New format option: `summary`:

image

Also, fixed json output.

## Testing

Covered with tests.

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/cli/src/commands/lint.ts [hunk 1]
```diff
@@ -127,6 +127,11 @@ function lintConfigCallback(argv: LintOptions, version: string) {
     return;
   }
 
+  if (argv.format === 'json') {
```
Comment (tatomyr) [inline packages/cli/src/commands/lint.ts:130]: Is this part covered with a test? If not, it's better to add one.

File: packages/core/src/format/format.ts [hunk 2]
```diff
@@ -257,6 +266,26 @@ export function formatProblems(
...
+function formatSummary(problems: NormalizedProblem[]): void {
+  const counts: Record<string, { count: number; severity: ProblemSeverity }> = {};
+  for (const problem of problems) {
+    counts[problem.ruleId] = counts[problem.ruleId] || { count: 0, severity: problem.severity };
+    counts[problem.ruleId].count++;
+  }
+  const sorted = Object.entries(counts).sort((a, b) => {
+    const severityDiff = severityToNumber(a[1].severity) - severityToNumber(b[1].severity);
+    return severityDiff || b[1].count - a[1].count;
+  });
```
Comment (tatomyr) [inline packages/core/src/format/format.ts:278]:
```suggestion
  const sorted = Object.entries(counts).sort(([, a], [, b]) => {
    const severityDiff = severityToNumber(a.severity) - severityToNumber(b.severity);
    return severityDiff || b.count - a.count;
  });
```

**Review by tatomyr — APPROVED** [review]
LGTM. 
Left a couple of optional suggestions.

File: packages/cli/src/commands/lint.ts (hunk 1 above)
Comment (RomanHotsiy) [reply packages/cli/src/commands/lint.ts:130]: Added an e2e test.

File: packages/core/src/format/format.ts (hunk 2 above)
Comment (RomanHotsiy) [reply packages/core/src/format/format.ts]: Thanks! 👍  Applied.

**Comment by jeremyfiel** [comment]
please update `lint` command docs with examples for each type of `format`

https://redocly.com/docs/cli/commands/lint/#format

**Pull request merged**